http_handler.h

```diff
@@ -12,10 +14,20 @@ struct request;
 class handler {
 public:
 
+    handler(std::string base_url) { this->base_url = base_url;};
+
     // Returns a response to the given request
     virtual response handle_request(const request& req) = 0;
+
+
+
+    std::string base_url;
```

Comment:
Lots of unnecessary newlines as well as no comment after the base_url member variable